src/sql/parts/profiler/editor/profilerInput.ts

```diff
@@ -123,7 +125,20 @@ export class ProfilerInput extends EditorInput implements IProfilerSession {
 		return this._state;
 	}
 
+	public onSessionStopped(notification: sqlops.ProfilerSessionStoppedParams) {
+		this._notificationService.error(nls.localize("profiler.sessionStopped", "Session stopped unexpectedly on the server."));
+		this.state.change({
+			isStopped: true,
+			isPaused: false,
+			isRunning: false
+		});
+	}
+
 	public onMoreRows(eventMessage: sqlops.ProfilerSessionEvents) {
+		if (eventMessage.eventsLost){
+			this._notificationService.warn(nls.localize("profiler.eventsLost", "Some events may have been lost."));
```

Comment:
Maybe the text such as the following.  "The XEvent Profiler session for {connection_servername_here} has lost events." is more descriptive.

The issues with current text are (1) it says "may have been" when shouldn't we know if the events were actually lost? (2) the message doesn't identify which profiling session lost events in case notification comes for profiling session that isn't the active document (3) message doesn't make clear it's related to profiler which could be confusing if profiler isn't active document.

Reply:
I totally agree on points 2 & 3, I'll make those changes. As for knowing if events were lost, all we can tell is that we can't find the last event we profiled, so the ring buffer may have filled up and overwritten old events that are now lost. I phrased it as "may" because there's the chance that the last overwritten event was the last seen event, and we haven't actually lost anything. It's probably a safe assumption that in all of these cases we will have actually lost events, so I can change that.

Reply:
@madelinemacd I left a comment on the Tools Service PR for item 1.  I think there is a way to use event sequence numbers to accurate detect lost events.  Let me know if you can't find the ID I'm referring to.  This isn't critical for this PR, but we should try to use that detection mechanism if possible.

Reply:
@kburtram Event sequence numbers hadn't even occurred to me, but that's a really simple way of doing it! I've found that ID, thanks!